src/os/bluestore/BlueStore.cc

```diff
@@ -18049,18 +18051,13 @@ int BlueStore::_setattr(TransContext *txc,
 	   << " " << name << " (" << val.length() << " bytes)"
 	   << dendl;
   int r = 0;
-  if (!val.length()) {
-    auto& b = o->onode.attrs[name.c_str()] = bufferptr("", 0);
```

Comment:
Case for `val.length()` is important.
`rebuild()` clears buffers and `front()` returns invalid reference I think.